### ISSUE TITLE
Perl5 configure does not run when LC_ALL=tr_TR

### ISSUE THREAD

**Issue by p5pRT** [Perl/perl5#6816]
Migrated from [rt.perl.org#24121](https://rt-archive.perl.org/perl5/Ticket/Display.html?id=24121) (status was 'resolved')

Searchable as RT24121$

**Comment by p5pRT** [comment]
### From ismail.donmez@boun.edu.tr

#### Created by ismail.donmez@boun.edu.tr


If your LC\_ALL=tr\_TR or similar \( like tr\_TR\.UTF\-8 \) perl5 \( 5\.8\.\{0\,1\} \) exits with

Good\, your tr supports \[&#8203;:lower&#8203;:\] and \[&#8203;:upper&#8203;:\] to convert case\.
I don't know how to translate letters from upper to lower case\.
Your tr is not acting any way I know of\.

This is because toupper\(i\) \!= I in Turkish but its big i with a dot above\.

<details><summary>Perl Info</summary>

```

Flags:
    category=install
    severity=medium

Site configuration information for perl v5.8.1:

Configured by root at Mon Sep 29 19:36:06 EEST 2003.

Summary of my perl5 (revision 5.0 version 8 subversion 1) configuration:
  Platform:
    osname=linux, osvers=2.6.0-test6-mm1, archname=i686-linux
    uname='linux southpark 2.6.0-test6-mm1 #9 mon sep 29 19:22:40 eest 2003 i686 unknown unknown gnulinux '
    config_args='-de -Dprefix=/usr -Dcccdlflags=-fPIC'
    hint=recommended, useposix=true, d_sigaction=define
    usethreads=undef use5005threads=undef useithreads=undef usemultiplicity=undef
    useperlio=define d_sfio=undef uselargefiles=define usesocks=undef
    use64bitint=undef use64bitall=undef uselongdouble=undef
    usemymalloc=n, bincompat5005=undef
  Compiler:
    cc='cc', ccflags ='-fno-strict-aliasing -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64',
    optimize='-O3',
    cppflags='-fno-strict-aliasing'
    ccversion='', gccversion='3.3', gccosandvers=''
    intsize=4, longsize=4, ptrsize=4, doublesize=8, byteorder=1234
    d_longlong=define, longlongsize=8, d_longdbl=define, longdblsize=12
    ivtype='long', ivsize=4, nvtype='double', nvsize=8, Off_t='off_t', lseeksize=8
    alignbytes=4, prototype=define
  Linker and Libraries:
    ld='cc', ldflags =''
    libpth=/lib /usr/lib
    libs=-lnsl -lndbm -ldl -lm -lcrypt -lutil -lc
    perllibs=-lnsl -ldl -lm -lcrypt -lutil -lc
    libc=/lib/libc-2.3.2.so, so=so, useshrplib=false, libperl=libperl.a
    gnulibc_version='2.3.2'
  Dynamic Linking:
    dlsrc=dl_dlopen.xs, dlext=so, d_dlsymun=undef, ccdlflags='-rdynamic'
    cccdlflags='-fPIC', lddlflags='-shared'

Locally applied patches:
    


@INC for perl v5.8.1:
    /usr/lib/perl5/5.8.1/i686-linux
    /usr/lib/perl5/5.8.1
    /usr/lib/perl5/site_perl/5.8.1/i686-linux
    /usr/lib/perl5/site_perl/5.8.1
    /usr/lib/perl5/site_perl
    .


Environment for perl v5.8.1:
    HOME=/home/cartman
    LANG (unset)
    LANGUAGE (unset)
    LC_ALL=tr_TR.UTF-8
    LD_LIBRARY_PATH (unset)
    LOGDIR (unset)
    PATH=~/cxoffice/bin:/opt/apache/bin:/opt/mysql/bin:/usr/batik:/opt/kde/qt/bin:/usr/java/bin:/usr/local/ssl/bin:/usr/acrobat/bin:/opt/kde/bin:/usr/local/bin:/usr/bin:/bin:/usr/X11R6/bin:/usr/games:/usr/Acrobat/bin:.:/sbin
    PERL_BADLANG (unset)
    SHELL=/bin/bash

```
</details>

**Comment by p5pRT** [comment]
### From @rgs

ismail donmez \(via RT\) wrote&#8203;:
> If your LC\_ALL=tr\_TR or similar \( like tr\_TR\.UTF\-8 \) perl5 \( 5\.8\.\{0\,1\} \) exits with
> 
> Good\, your tr supports \[&#8203;:lower&#8203;:\] and \[&#8203;:upper&#8203;:\] to convert case\.
> I don't know how to translate letters from upper to lower case\.
> Your tr is not acting any way I know of\.
> 
> This is because toupper\(i\) \!= I in Turkish but its big i with a dot above\.

Thanks for your report\.

The best fix is probably to replace the various "ij" and "IJ" in the
Configure tests for this by "rs" etc\. \(there's also a discontinuity
in EBCDIC between R and S\, as for I and J\.\)

\[Merijn\, I can provide a patch if you want\.\]

**Comment by p5pRT** [comment]
### From @tux

On Wed 08 Oct 2003 00&#8203;:01\, Rafael Garcia\-Suarez \<rgarciasuarez@&#8203;free\.fr> wrote&#8203;:
> ismail donmez \(via RT\) wrote&#8203;:
> > If your LC\_ALL=tr\_TR or similar \( like tr\_TR\.UTF\-8 \) perl5 \( 5\.8\.\{0\,1\} \) exits with
> > 
> > Good\, your tr supports \[&#8203;:lower&#8203;:\] and \[&#8203;:upper&#8203;:\] to convert case\.
> > I don't know how to translate letters from upper to lower case\.
> > Your tr is not acting any way I know of\.
> > 
> > This is because toupper\(i\) \!= I in Turkish but its big i with a dot above\.
> 
> Thanks for your report\.
> 
> The best fix is probably to replace the various "ij" and "IJ" in the
> Configure tests for this by "rs" etc\. \(there's also a discontinuity
> in EBCDIC between R and S\, as for I and J\.\)
> 
> \[Merijn\, I can provide a patch if you want\.\]

Yes please\. Code speaks better than text\.

I'm a bit late with this\, aren't I?
Is it still an issue?

\-\- 
H\.Merijn Brand        Amsterdam Perl Mongers \(http&#8203;://amsterdam\.pm\.org/\)
using perl\-5\.6\.1\, 5\.8\.3\, & 5\.9\.x\, and 809 on  HP\-UX 10\.20 & 11\.00\, 11i\,
&nbsp;   AIX 4\.3\, SuSE 9\.0\, and Win2k\.           http&#8203;://www\.cmve\.net/~merijn/
http&#8203;://archives\.develooper\.com/daily\-build@&#8203;perl\.org/   perl\-qa@&#8203;perl\.org
send smoke reports to&#8203;: smokers\-reports@&#8203;perl\.org\, QA&#8203;: http&#8203;://qa\.perl\.org

**Comment by p5pRT** [comment]
p5p@spam.wizbit.be - Status changed from 'open' to 'resolved'

**Comment by p5pRT** [comment]
### From @smpeters

On Tue Apr 13 06&#8203;:04&#8203;:07 2004\, hmbrand wrote&#8203;:
> On Wed 08 Oct 2003 00&#8203;:01\, Rafael Garcia\-Suarez \<rgarciasuarez@&#8203;free\.fr>
> wrote&#8203;:
> > ismail donmez \(via RT\) wrote&#8203;:
> > > If your LC\_ALL=tr\_TR or similar \( like tr\_TR\.UTF\-8 \) perl5 \(
> 5\.8\.\{0\,1\} \) exits with
> > >
> > > Good\, your tr supports \[&#8203;:lower&#8203;:\] and \[&#8203;:upper&#8203;:\] to convert case\.
> > > I don't know how to translate letters from upper to lower case\.
> > > Your tr is not acting any way I know of\.
> > >
> > > This is because toupper\(i\) \!= I in Turkish but its big i with a
> dot above\.
> >
> > Thanks for your report\.
> >
> > The best fix is probably to replace the various "ij" and "IJ" in the
> > Configure tests for this by "rs" etc\. \(there's also a discontinuity
> > in EBCDIC between R and S\, as for I and J\.\)
> >
> > \[Merijn\, I can provide a patch if you want\.\]
> 
> Yes please\. Code speaks better than text\.
> 
> I'm a bit late with this\, aren't I?
> Is it still an issue?
> 


This was fixed with change \#22696\.

Thanks for the report
